admin/jobs/river/biller_event_handlers.go

```diff
@@ -0,0 +1,286 @@
+
+type InvoicePaymentFailedWorker struct {
+	river.WorkerDefaults[InvoicePaymentFailedArgs]
+	admin *admin.Service
+}
+
+func (w *InvoicePaymentFailedWorker) Work(ctx context.Context, job *river.Job[InvoicePaymentFailedArgs]) error {
+	org, err := w.admin.DB.FindOrganizationForBillingCustomerID(ctx, job.Args.BillingCustomerID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			// org got deleted, ignore
+			return nil
+		}
+		return fmt.Errorf("failed to find organization of billing customer id %q: %w", job.Args.BillingCustomerID, err)
+	}
+
+	// schedule a job to check if the invoice is paid after end of grace period
+	gracePeriodEndDate := job.Args.FailedAt.Truncate(24*time.Hour).AddDate(0, 0, gracePeriodDays)
+	j, err := w.admin.Jobs.InvoicePaymentFailedGracePeriodCheck(ctx, org.ID, job.Args.InvoiceID, gracePeriodEndDate)
+	if err != nil {
+		return fmt.Errorf("failed to schedule invoice payment failed grace period check job: %w", err)
+	}
+
+	be, err := w.admin.DB.FindBillingErrorByType(ctx, org.ID, database.BillingErrorTypeInvoicePaymentFailed)
+	if err != nil {
+		if !errors.Is(err, database.ErrNotFound) {
+			return fmt.Errorf("failed to find billing errors: %w", err)
+		}
+	}
+	var metadata *database.BillingErrorMetadataInvoicePaymentFailed
+	if be != nil {
+		metadata = be.Metadata.(*database.BillingErrorMetadataInvoicePaymentFailed)
+	} else {
+		metadata = &database.BillingErrorMetadataInvoicePaymentFailed{
+			Invoices: make(map[string]database.InvoicePaymentFailedMeta),
+		}
+	}
+
+	metadata.Invoices[job.Args.InvoiceID] = database.InvoicePaymentFailedMeta{
+		ID:                  job.Args.InvoiceID,
+		Number:              job.Args.InvoiceNumber,
+		URL:                 job.Args.InvoiceURL,
+		Amount:              job.Args.Amount,
+		Currency:            job.Args.Currency,
+		DueDate:             job.Args.DueDate,
+		FailedOn:            job.Args.FailedAt,
+		GracePeriodEndJobID: j.ID,
+	}
+
+	// insert billing error
+	_, err = w.admin.DB.UpsertBillingError(ctx, &database.UpsertBillingErrorOptions{
+		OrgID:     org.ID,
+		Type:      database.BillingErrorTypeInvoicePaymentFailed,
+		Metadata:  &database.BillingErrorMetadataInvoicePaymentFailed{Invoices: metadata.Invoices},
+		EventTime: job.Args.FailedAt,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to add billing error: %w", err)
+	}
+
+	// send email
+	err = w.admin.Email.SendInformational(&email.Informational{
+		ToEmail: org.BillingEmail,
+		ToName:  org.Name,
+		Subject: "Your invoice payment failed",
+		Title:   "",
+		Body:    template.HTML(fmt.Sprintf("Your invoice payment failed, please visit the billing portal to fix issues or contact support to continue using Rill. Your projects will be hibernated on %s UTC, if invoice still not paid.", gracePeriodEndDate.Format("2006-01-02"))),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to send invoice payment failed email for org %q: %w", org.Name, err)
+	}
```

Comment:
This should be wrapped in a function in the `runtime/pkg/email` package, something like `Email.SendPaymentFailed` or something like that. The goal is that all email text is contained in the `email` package so that it's easier for the product folks to audit and edit.